File: python/program/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    def __init__(self, plat, merk, tahun, warna, jenis_motor, tangki):
        super().__init__(plat, merk, tahun, warna)
        self.__jenis_motor = jenis_motor
        self.__tangki = tangki

    def get_jenis_motor(self):
        return self.__jenis_motor

    def set_jenis_motor(self, jenis_motor):
        self.__jenis_motor = jenis_motor

    def get_tangki(self):
        return self.__tangki

    def set_tangki(self, tangki):
        self.__tangki = tangki

File: LP4DPBO2024C1/program/Motorcycle.cpp


#include <bits/stdc++.h>
#include "Vehicle.h"

using namespace std;

class Motorcycle : public Vehicle
{
private:
    string jenis_motor;
    int kapasitas_tangki;

public:
    Motorcycle()
    {

    }

    Motorcycle(string plat, string merk, int tahun, string warna, string jenis_motor, int kapasitas_tangki)
        : Vehicle(plat, merk, tahun, warna), jenis_motor(jenis_motor), kapasitas_tangki(kapasitas_tangki)
    {

    }

    void setJenisMotor(string jenis_motor)
    {
        this->jenis_motor = jenis_motor;
    }

    string getJenisMotor() const
    {
        return jenis_motor;
    }

    void setKapasitasTangki(int kapasitas_tangki)
    {
        this->kapasitas_tangki = kapasitas_tangki;
    }

    int getKapasitasTangki() const
    {
        return kapasitas_tangki;
    }

    ~Motorcycle()
    {

    }
};
